io/src/pcd_io.cpp

```diff
@@ -254,7 +254,7 @@ pcl::PCDReader::readHeader (std::istream &fs, pcl::PCLPointCloud2 &cloud,
           cloud.fields[i].offset = offset;
           int col_count;
           sstream >> col_count;
-          if (col_count < 1)
+          if (col_count < 1 && cloud.fields[i].name != "_")
```

Comment:
Instead of moving ahead with a 0 count, should we be using erase+remove on `cloud.fields[i]`? (either here or after reading the header)